clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventHandler.java

```diff
@@ -0,0 +1,122 @@
+                                   final Supplier<ApplicationEventProcessor> applicationEventProcessorSupplier,
+                                   final Supplier<NetworkClientDelegate> networkClientDelegateSupplier,
+                                   final Supplier<RequestManagers> requestManagersSupplier) {
+        this.log = logContext.logger(ApplicationEventHandler.class);
+        this.applicationEventQueue = applicationEventQueue;
+        this.backgroundThread = new DefaultBackgroundThread(time,
+                logContext,
+                applicationEventProcessorSupplier,
+                networkClientDelegateSupplier,
+                requestManagersSupplier);
+        this.backgroundThread.start();
+    }
+
+    /**
+     * Add an {@link ApplicationEvent} to the handler.
+     *
+     * @param event An {@link ApplicationEvent} created by the application thread
+     */
+    public void add(final ApplicationEvent event) {
+        Objects.requireNonNull(event, "ApplicationEvent provided to add must be non-null");
+        log.trace("Enqueued event: {}", event);
+        backgroundThread.wakeup();
+        applicationEventQueue.add(event);
+    }
+
+    /**
+     * Add a {@link CompletableApplicationEvent} to the handler. The method blocks waiting for the result, and will
+     * return the result value upon successful completion; otherwise throws an error.
+     *
+     * <p/>
+     *
+     * See {@link CompletableApplicationEvent#get(Timer)} and {@link Future#get(long, TimeUnit)} for more details.
+     *
+     * @param event A {@link CompletableApplicationEvent} created by the polling thread.
+     * @param timer Timer for which to wait for the event to complete
+     * @return      Value that is the result of the event
+     * @param <T>   Type of return value of the event
+     */
+    public <T> T addAndGet(final CompletableApplicationEvent<T> event, final Timer timer) {
+        Objects.requireNonNull(event, "CompletableApplicationEvent provided to addAndGet must be non-null");
+        Objects.requireNonNull(timer, "Timer provided to addAndGet must be non-null");
+        add(event);
+        return event.get(timer);
+    }
+
+    @Override
+    public void close() {
+        close(Duration.ofMillis(Long.MAX_VALUE));
+    }
+
+    public void close(final Duration timeout) {
+        Objects.requireNonNull(timeout, "Duration provided to close must be non-null");
+
+        closer.close(
+                () ->  {
+                    log.info("Closing the consumer application event handler");
+
+                    try {
+                        long timeoutMs = timeout.toMillis();
+
+                        if (timeoutMs < 0)
+                            throw new IllegalArgumentException("The timeout cannot be negative.");
+
+                        backgroundThread.close();
+                        log.info("The consumer application event handler was closed");
```

Comment:
we are logging debug for the request manager but info here.  Just want to make sure we are consistent with the logging level.

Reply:
Changed to debug